flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -138,6 +139,8 @@ public class ExecutionConfig implements Serializable, Archiveable<ArchivedExecut
 
     private boolean isLatencyTrackingConfigured = false;
 
+    private TernaryBoolean changelogEnabled = TernaryBoolean.UNDEFINED;
+
```

Comment:
Have you considered moving the configuration to `JobInformation.jobConfiguration` instead of `ExecutionConfig`?

I think it would be easier for
- FLINK-26372 because `StateChangelogStorageFactory` requires multiple parameters
- and probably for FLINK-27692 if there will be more conifg parameters

WDYT?

cc: @zoltar9264

Reply:
Yes, but there are two things to consider:
1. `JobInformation` is in `flink-runtime` module. Moving the configuration to `JobInformation.jobConfiguration` is not much different from the [current situation](https://github.com/apache/flink/pull/19907#discussion_r902485780) (`changelogEnabled ` is in `TaskInformation.taskConfiguration`). Do you mean moving the parsing to `JobInformation`? something like:

```java
public class JobInformation implements Serializable {
private static final String ENABLE_CHANGE_LOG_STATE_BACKEND = "enablechangelog";
  /** Configuration of the job. */
 private final Configuration jobConfiguration;
...
 public void setChangelogStateBackendEnabled(TernaryBoolean enabled) {
        try {
            InstantiationUtil.writeObjectToConfig(
                    enabled, this.jobConfiguration, ENABLE_CHANGE_LOG_STATE_BACKEND);
        } catch (Exception e) {
            throw new StreamTaskException(
                    "Could not serialize change log state backend enable flag.", e);
        }
    }
public TernaryBoolean isChangelogStateBackendEnabled(ClassLoader cl) {
        try {
            return InstantiationUtil.readObjectFromConfig(
                    this.jobConfiguration, ENABLE_CHANGE_LOG_STATE_BACKEND, cl);
        } catch (Exception e) {
            throw new StreamTaskException(
                    "Could not instantiate change log state backend enable flag.", e);
        }
    }
...
}
```
2. `BootstrapTransformation` in `flink-state-processing-api` also needs `enablechangelog` configuration, if move it  to `JobInformation.jobConfiguration`, do we need to keep a copy of this configuration in `flink-streaming-java`?

Reply:
I meant making `JobInformation.jobConfiguration` the **container** for changelog configuration; but having the constants (and the parsing code) in `flink-core`. That would be similar to `DistributedCache`.

`BootstrapTransformation` would be able to access the config too.

WDYT?

Reply:
The `periodicMaterializeIntervalMillis` and `materializationMaxAllowedFailures` associated with changelog are also in `ExecutionConfig`,  and I can't find a good class or a new class to put the constants (and the parsing code) in `flink-core`, do you have any suggestion?